src/navigation/failure_identification/failure_identification.py

```diff
@@ -0,0 +1,216 @@
+            [nav_status_sub, odometry_sub],
+            10,
+            1.0,
+            allow_headerless=True,
+        )
+        ts.registerCallback(self.update)
+
+        self.stuck_publisher = rospy.Publisher("/nav_stuck", Bool, queue_size=1)
+        position_variables = ["x", "y", "z"]
+        rotation_variables = [f"rot_{comp}" for comp in ["x", "y", "z", "w"]]
+        velocity_variables = ["linear_velocity", "angular_velocity"]
+        # wheel_effort_variables = [f"wheel_{wheel_num}_effort" for wheel_num in range(6)]
+        # wheel_velocity_variables = [f"wheel_{wheel_num}_velocity" for wheel_num in range(6)]
+        command_variables = ["cmd_vel_x", "cmd_vel_twist"]
+        self.data_collecting_mode = True
+        self.actively_collecting = True
+        self.cur_cmd = Twist()
+        self.cur_stuck = False
+        self.cols = (
+            ["time", "stuck"]
+            + position_variables
+            + rotation_variables
+            + velocity_variables
+            # + wheel_effort_variables
+            # + wheel_velocity_variables
+            + command_variables
+        )
+        print(self.cols)
+        self._df = pd.DataFrame(columns=self.cols)
+        self.left_pointer = 0
+        self.right_pointer = 0
+        self.row_counter = 0
+        self.watchdog = WatchDog(self)
+        self.path_name = None
+
+    def write_to_csv(self):
+        """
+        Writes the data frame to a csv file marked with timestamp
+        """
+        home = Path.home()
+        # path = os.path.join(home, "catkin_ws/src/mrover-workspace/src/navigation/failure_data")
+        path = home / "catkin_ws/src/mrover/src/failure_data"
+        path.mkdir(exist_ok=True)
+        # now = datetime.datetime.now()
+        # day = now.strftime("%m%d%Y")
+        # hour = now.strftime("%H-%M-%S")
+        # time_stamp = day + "_" + hour
+
+        file_name = f"failure_data_{rospy.Time.now()}.csv"
+        # file_name = f"failure_data_{time_stamp}.csv"
+        path = path / file_name
+        self.path_name = path
+        self._df.to_csv(path)
+
+        # now = datetime.datetime.now()
+        # day = now.strftime("%m%d%Y")
+        # hour = now.strftime("%H-%M-%S")
+        # time_stamp = day + "_" + hour
+        # home = str(Path.home())
+        # folder = home + "/catkin_ws/src/mrover/failure_data"
+        # if not os.path.exists(folder):
+        #     os.makedirs(folder)
+        # file = folder + "/output_averaged_" + time_stamp + ".csv"
+
+        rospy.loginfo("===== failure data written to csv =====")
+
+    def stuck_button_update(self, stuck_button: Bool):
+        self.cur_stuck = stuck_button.data
+
+    def cmd_vel_update(self, cmd_vel: Twist):
+        self.cur_cmd = cmd_vel
+
+    # def update(self, nav_status: SmachContainerStatus, drive_status: MotorsStatus, odometry: Odometry):
+    def update(self, nav_status: SmachContainerStatus, odometry: Odometry):
```

Comment:
can you start subscribing to drive_status again? If you merge the most recent master this should work in sim now